pkg/web/oauthclient/callback.go

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauthclient
+
+import (
+	"net/http"
+
+	echo "github.com/labstack/echo/v4"
+)
+
+// HandleCallback is a handler that takes the auth code and exchanges it for the
+// access token.
+func (oc *OAuthClient) HandleCallback(c echo.Context) error {
+	if e := c.QueryParam("error"); e != "" {
+		return echo.NewHTTPError(http.StatusUnauthorized, c.QueryParam("error_description"))
```

Comment:
Perhaps we should make this a registered error instead of an echo error. Same for the ones below.